### PR TITLE
Refactor cleanup

### PR DESCRIPTION
**Pull request by LoRyder1**
@drmyersii Check out some of the changes I made!

### CODE REVIEW

File: lib/SnakesLadders.rb [hunk 1]
```diff
@@ -2,16 +2,15 @@
 require File.dirname(__FILE__) + '/Space'
 
 class SnakesLadders
-    # declare instance variables
-    @players
-    @spaces
-    @is_finished
```
Comment (LoRyder1) [inline lib/SnakesLadders.rb:8]: I got rid of the declarations here. Initializing declares the vars automatically

File: lib/Space.rb [hunk 2]
```diff
@@ -15,9 +11,11 @@ def initialize
     # transform the Space into a different type (snake or ladder)
     def transform type
         # ensure type is allowed
+        @type = type
         case type
-        when 'snake', 'ladder'
-            @type           = type
+        when 'snake'
+            @spaces_to_move = - rand(1..10)
```
Comment (LoRyder1) [inline lib/Space.rb:17]: allowed for negative spaces to move when type is snake

File: lib/SnakesLadders.rb [hunk 3]
```diff
@@ -23,46 +22,26 @@ def run
...
                 # check to see if the player has won the game
                 if (player.position == @spaces.length)
-                    puts ""
-                    puts "#{player.name} Wins!"
-                    puts ""
-
+                    puts (<<-WIN)
+                        #{player.name} Wins!
+                    
+                    WIN
```
Comment (LoRyder1) [inline lib/SnakesLadders.rb:44]: heredocs in action!

File: lib/SnakesLadders.rb [hunk 4]
```diff
@@ -23,46 +22,26 @@ def run
...
-                # get the player's roll
-                roll = gets.chomp.to_i
-
-                # make sure the roll is valid
-                while (roll < 1 || roll > 6)
-                    puts "Invalid Roll! Please try that again:"
-
-                    roll = gets.chomp.to_i
-                end
+                validate_roll
```
Comment (LoRyder1) [inline lib/SnakesLadders.rb:27]: having a good method name eliminates need for comment

File: lib/SnakesLadders.rb [hunk 5]
```diff
@@ -23,46 +22,26 @@ def run
...
-                    puts ""
-                    puts "You landed on a ladder! You get to move forward #{space.spaces_to_move} spaces!"
-                    puts ""
+                # set instance var of the Space the player landed on
+                @current_space = @spaces[player.position - 1]
 
-                    player.move(space.spaces_to_move)
-                elsif (space.type == 'snake')
-                    puts ""
-                    puts "You landed on a snake! You have to move back #{space.spaces_to_move} spaces!"
```
Comment (LoRyder1) [inline lib/SnakesLadders.rb:52]: code duplication here so I created output_snakeladder_msg method

**Comment by davidmyersdev** [comment]
Hey! Sorry about not messaging you back on this! I like the changes, and I see that you merged them in now. This kata is looking good!

**Pull request merged**